frontend/workflows/projectSelector/src/selector-reducer.tsx

```diff
@@ -63,6 +71,30 @@ const selectorReducer = (state: State, action: Action): State => {
           ),
         },
       };
+
+      if (action.payload.group !== Group.PROJECTS) {
+        return newState;
+      }
+
+      // check to see if we have project data for project (b/c we don't need to make an API call) and if so, update the state with
+      // its upstreams/downstreams
+      uniqueCustomProjects.forEach(v => {
+        if (v in newState.projectData) {
+          newState.projectData[v].dependencies.upstreams[TYPEURL]?.ids.forEach(upstreamDep => {
```

Comment:
i think this will work
```suggestion
          const { upstreams, downstreams } = newState.projectData[v].dependencies;
          upstreams[TYPEURL]?.ids.forEach(upstreamDep => {
```

if not at least do
```jsx
const dependencies = newState.projectData[v].dependencies;
dependencies.upstreams[TYPEURL]?.ids.forEach(upstreamDep => {
```

Reply:
nice thx! updated "hydrate_end" as well